sos/report/plugins/ceph_osd.py

```diff
@@ -31,62 +31,77 @@ class CephOSD(Plugin, RedHatPlugin, UbuntuPlugin):
-            "/run/ceph/**/ceph-osd*",
-            "/var/lib/ceph/**/kv_backend",
-            "/var/log/ceph/**/ceph-osd*.log",
-            "/var/log/ceph/**/ceph-volume*.log",
-        ])
-
-        self.add_cmd_output([
-            "ceph-disk list",
-            "ceph-volume lvm list"
-        ])
-
-        cmds = [
-           "bluestore bluefs available",
-           "config diff",
-           "config show",
-           "dump_blacklist",
-           "dump_blocked_ops",
-           "dump_historic_ops_by_duration",
-           "dump_historic_slow_ops",
-           "dump_mempools",
-           "dump_ops_in_flight",
-           "dump_op_pq_state",
-           "dump_osd_network",
-           "dump_reservations",
-           "dump_watchers",
-           "log dump",
-           "perf dump",
-           "perf histogram dump",
-           "objecter_requests",
-           "ops",
-           "status",
-           "version",
-        ]
-
-        self.add_cmd_output(
-            [f"ceph daemon {i} {c}" for i in self.get_socks() for c in cmds]
-        )
+        if not self.path_exists('/var/snap/microceph/common'):
+            self.add_file_tags({
+                "/var/log/ceph/(.*/)?ceph-(.*-)?osd.*.log": 'ceph_osd_log',
+            })
+
+            self.add_forbidden_path([
+                "/etc/ceph/*keyring*",
+                "/var/lib/ceph/**/*keyring*",
+                # Excludes temporary ceph-osd mount location like
+                # /var/lib/ceph/tmp/mnt.XXXX from sos collection.
+                "/var/lib/ceph/**/tmp/*mnt*",
+                "/etc/ceph/*bindpass*"
+            ])
+
+            # Only collect OSD specific files
+            self.add_copy_spec([
+                "/run/ceph/**/ceph-osd*",
+                "/var/lib/ceph/**/kv_backend",
+                "/var/log/ceph/**/ceph-osd*.log",
+                "/var/log/ceph/**/ceph-volume*.log",
+            ])
+
+            self.add_cmd_output([
+                "ceph-disk list",
+                "ceph-volume lvm list"
+            ])
+
+            cmds = [
+               "bluestore bluefs available",
+               "config diff",
+               "config show",
+               "dump_blacklist",
+               "dump_blocked_ops",
+               "dump_historic_ops_by_duration",
+               "dump_historic_slow_ops",
+               "dump_mempools",
+               "dump_ops_in_flight",
+               "dump_op_pq_state",
+               "dump_osd_network",
+               "dump_reservations",
+               "dump_watchers",
+               "log dump",
+               "perf dump",
+               "perf histogram dump",
+               "objecter_requests",
+               "ops",
+               "status",
+               "version",
+            ]
+
+            self.add_cmd_output(
+              [f"ceph daemon {i} {c}" for i in self.get_socks() for c in cmds]
+            )
+
+        else:
+            # Only collect microceph files, don't run any commands
```

Comment:
Why don't we want to run any commands? I would expect to still want all the OSD commands regardless of whether we are using microceph or not.

Reply:
in microceph, I can run the ceph commands (on the mon nodes) but not the ceph daemon commands on the OSD nodes that the OSD nodes run for ceph_osd collections. 

~~~
ubuntu@demonax:~$ sudo -s
root@demonax:/home/ubuntu# ceph -s
  cluster:
    id:     fea9b2ee-20f0-45c1-838e-03bc1d6667bf
    health: HEALTH_OK
 
  services:
    mon: 1 daemons, quorum demonax (age 5w)
    mgr: demonax(active, since 5w)
    osd: 3 osds: 3 up (since 5w), 3 in (since 5w)
 
  data:
    pools:   1 pools, 1 pgs
    objects: 2 objects, 577 KiB
    usage:   240 MiB used, 11 TiB / 11 TiB avail
    pgs:     1 active+clean
 
root@demonax:/home/ubuntu# ps -efa | grep osd
root       34627       1  0 Jun02 ?        00:00:00 /bin/sh /snap/microceph/338/commands/osd.start
root       36645       1  0 Jun02 ?        02:15:10 ceph-osd --cluster ceph --id 0
root       38275       1  0 Jun02 ?        02:17:26 ceph-osd --cluster ceph --id 1
root       39929       1  0 Jun02 ?        02:16:48 ceph-osd --cluster ceph --id 2
root      195108  195052  0 09:23 pts/1    00:00:00 grep --color=auto osd
root@demonax:/home/ubuntu# ceph daemon osd.0 help
Can't get admin socket path: "ceph-conf" not found
~~~

Reply:
I'll submit a new PR later to add the ceph daemon command collections once  https://github.com/canonical/microceph/issues/160 gets resolved.